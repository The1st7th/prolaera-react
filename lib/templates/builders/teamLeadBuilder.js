'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _extends =
  Object.assign ||
  function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactHtmlEmail = require('react-html-email');

var _footer = require('../../components/footer');

var _footer2 = _interopRequireDefault(_footer);

var _header = require('../../components/header');

var _header2 = _interopRequireDefault(_header);

var _subHeader = require('../../components/subHeader');

var _subHeader2 = _interopRequireDefault(_subHeader);

var _teamInfo = require('../../components/team/teamInfo');

var _teamInfo2 = _interopRequireDefault(_teamInfo);

var _templateCSS = require('../templateCSS.js');

var _templateCSS2 = _interopRequireDefault(_templateCSS);

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

function _asyncToGenerator(fn) {
  return function() {
    var gen = fn.apply(this, arguments);
    return new Promise(function(resolve, reject) {
      function step(key, arg) {
        try {
          var info = gen[key](arg);
          var value = info.value;
        } catch (error) {
          reject(error);
          return;
        }
        if (info.done) {
          resolve(value);
        } else {
          return Promise.resolve(value).then(
            function(value) {
              step('next', value);
            },
            function(err) {
              step('throw', err);
            }
          );
        }
      }
      return step('next');
    });
  };
}

var teamLeadBuilder = (function() {
  var _ref = _asyncToGenerator(
    /*#__PURE__*/ regeneratorRuntime.mark(function _callee(completeTeam, imageUrl) {
      var Header, adminProfileId;
      return regeneratorRuntime.wrap(
        function _callee$(_context) {
          while (1) {
            switch ((_context.prev = _context.next)) {
              case 0:
                _context.prev = 0;
                _context.next = 3;
                return (0, _header2.default)(imageUrl);

              case 3:
                Header = _context.sent;
                adminProfileId = 'adminProfileId';
                return _context.abrupt(
                  'return',
                  (0, _reactHtmlEmail.renderEmail)(
                    _react2.default.createElement(
                      _reactHtmlEmail.Email,
                      { title: "You've been added as a team leader", headCSS: _templateCSS2.default },
                      _react2.default.createElement(Header, {
                        text: 'View Team',
                        link: 'https://app.prolaera.com/#/admin/' + adminProfileId + '/dashboard'
                      }),
                      _react2.default.createElement(_subHeader2.default, {
                        subText: 'New team',
                        header: 'Leader',
                        message: 'You have been added to ' + completeTeam.name + ' as team leader.'
                      }),
                      _react2.default.createElement(
                        _teamInfo2.default,
                        _extends({}, completeTeam, { buttonText: 'View Team', buttonLink: adminProfileId })
                      ),
                      _react2.default.createElement(_footer2.default, null)
                    )
                  )
                );

              case 8:
                _context.prev = 8;
                _context.t0 = _context['catch'](0);
                throw _context.t0;

              case 11:
              case 'end':
                return _context.stop();
            }
          }
        },
        _callee,
        undefined,
        [[0, 8]]
      );
    })
  );

  return function teamLeadBuilder(_x, _x2) {
    return _ref.apply(this, arguments);
  };
})();

exports.default = teamLeadBuilder;
