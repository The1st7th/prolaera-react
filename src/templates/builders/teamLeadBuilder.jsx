import React from 'react';
import Footer from '../../components/footer';
import builderHeader from '../../components/header';
import SubHeader from '../../components/subHeader';
import TeamInfo from '../../components/team/teamInfo';
import css from '../templateCSS.js';
import { renderToStaticMarkup } from 'react-dom/server';
import Email from '../../components/newemail';

const teamLeadBuilder = async (completeTeam, imageUrl) => {
  try {
    const Header = await builderHeader(imageUrl);
    const adminProfileId = 'adminProfileId';
    return renderEmail(
      <Email title="You've been added as a team leader" headCSS={css}>
        <Header text={'View Team'} link={'https://app.prolaera.com/#/admin/' + adminProfileId + '/dashboard'} />
        <SubHeader subText={"You've Been Added As"} header={'Team Leader'} message={'For ' + completeTeam.name} />
        <TeamInfo {...completeTeam} buttonText={'View Team'} buttonLink={adminProfileId} />
        <Footer />
      </Email>
    );
  } catch (error) {
    throw error;
  }
};
function renderEmail(emailComponent) {
  var doctype =
    '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">';
  return doctype + renderToStaticMarkup(emailComponent);
}
export default teamLeadBuilder;
