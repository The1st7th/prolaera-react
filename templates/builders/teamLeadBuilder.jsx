import React from 'react';
import { Email, renderEmail } from 'react-html-email';
import Footer from '../../components/footer';
import builderHeader from '../../components/header';
import SubHeader from '../../components/subHeader';
import TeamInfo from '../../components/team/teamInfo';
import css from '../templateCSS.js';

const teamLeadBuilder = async (completeTeam, imageUrl) => {
  try {
    const Header = await builderHeader(imageUrl);
    const adminProfileId = 'adminProfileId';
    return renderEmail(
      <Email title="You've been added as a team leader" headCSS={css}>
        <Header text={'View Team'} link={'https://app.prolaera.com/#/admin/' + adminProfileId + '/dashboard'} />
        <SubHeader
          subText={'New team'}
          header={'Leader'}
          message={'You have been added to ' + completeTeam.name + ' as team leader.'}
        />
        <TeamInfo {...completeTeam} buttonText={'View Team'} buttonLink={adminProfileId} />
        <Footer />
      </Email>
    );
  } catch (error) {
    throw error;
  }
};

export default teamLeadBuilder;
