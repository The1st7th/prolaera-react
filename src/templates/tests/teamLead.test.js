import renderer from 'react-test-renderer';
import writeFile from '../../helpers/writeFileHelper';
import teamLeadBuilder from '../builders/teamLeadBuilder';
import inlineCss from 'inline-css';
var request = require('request');

const _sendEmail = (template, emails) => {
  return new Promise((resolve, reject) => {
    request(
      {
        url: 'https://api-dev.prolaera.com/v1/mailer',
        method: 'PUT',
        json: {
          template,
          emails
        }
      },
      (error, response, body) => {
        if (error) {
          console.error(error);
          return reject(error);
        }
        resolve({
          response,
          body
        });
      }
    );
  });
};
describe('teamLead Email', () => {
  const logoUrl = 'https://assets.prolaera.com/prolaeraLogo_fullText.png';
  const team = {
    name: 'Test CPA Team',
    description: 'This team of CPAs handles all of the tax preparation in the firm.',
    buttonText: 'Test Button Text',
    buttonLink: 'www.testButtonLink.com'
  };
  it('returns teamLead email html', async () => {
    const emailHtml = await teamLeadBuilder(team, logoUrl);
    expect(emailHtml).toBeDefined();
  });

  it('checks snapshot', async () => {
    const emailHtml = await teamLeadBuilder(team, logoUrl);
    const emailComponent = renderer.create(emailHtml);
    const teamLeadJson = emailComponent.toJSON();
    expect(teamLeadJson).toMatchSnapshot();
  });

  it('writes an html file', async () => {
    const email = await teamLeadBuilder(team, logoUrl);
    const newEmail = await inlineCss(email, {
      url: ' ',
      preserveMediaQueries: true,
      applyWidthAttributes: true,
      applyTableAttributes: true
    });
    await _sendEmail(newEmail, ['aflupton@gmail.com', 'andrew@prolaera.onmicrosoft.com']);
    const saved = await writeFile(newEmail, 'teamLeadTest.html');
    expect(saved).toEqual(true);
  });
});
