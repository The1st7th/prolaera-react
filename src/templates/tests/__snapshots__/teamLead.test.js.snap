// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teamLead Email checks snapshot 1`] = `
"<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Strict//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\\"><html lang=\\"en\\" xmlns=\\"http://www.w3.org/1999/xhtml\\"><head><meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=utf-8\\"/><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"/><title>You&#x27;ve been added as a team leader</title><style type=\\"text/css\\">body {
	background-color: #F7F7F7;
	font-family: source-sans-pro, sans-serif;
}

#subHeaderWrapper {
	padding: 20px;
}

.innerHtmlStyles {
	margin-top: 10px;
}
.innerHtmlStyles p, .innerHtmlStyles ol, .innerHtmlStyles ul  {
	margin-top: 10px;
}

@media
only screen and (max-width: 600px),
(min-device-width: 600px) and (max-device-width: 1024px)  {

	/* Force table to not be like tables anymore */
	table, thead, tbody, th, td, tr {
		display: block;
	}

	.emailHeader td {
		width: 100%;
	}

	.emailHeader td img {
		margin-left: auto;
		margin-right: auto;
	} 

	.emailFooter {
		width: 100%;
		color: #F7F7F7;
		margin-right: auto;
		margin-left: auto;
		font-size: 10pt;
	}
	.emailFooter td {
		padding: 0px 10px 5px 10px;
		font-size: 10pt;
	}

	#subHeaderWrapper {
		width: 100%;
		padding: 0px;
	}

	.report td {
		/* Behave  like a &quot;row&quot; */
		border: none;
		border-bottom: 1px solid #eee;
    	position: relative;
    	text-align: center;
	}

	.report td:before {
		/* Now like a table header */
		position: absolute;
		/* Top/left values mimic padding */
		top: 6px;
		left: 6px;
		width: 45%;
		padding-right: 10px;
		white-space: nowrap;
	}
	
	.customButton:hover {
		color:black;
		
	}


/*End of media query*/
}</style></head><body style=\\"width:100%;margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%\\"><table width=\\"100%\\" height=\\"100%\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" align=\\"left\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\" valign=\\"top\\"><table width=\\"600\\" align=\\"center\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><table class=\\"emailHeader\\" align=\\"center\\" width=\\"100%\\" style=\\"background-color:#F7F7F7\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\"><table align=\\"center\\" cellSpacing=\\"0\\" style=\\"padding:15px;max-width:584px;margin-top:25px;margin-bottom:20px\\"><tr align=\\"center\\"><td width=\\"292px\\"><img alt=\\"logo\\" src=\\"https://assets.prolaera.com/prolaeraLogo_fullText.png\\" width=\\"199.00305623471883\\" height=\\"47\\" style=\\"display:block;outline:none;border:none;text-decoration:none\\"/></td></tr></table></td></tr></tbody></table><div id=\\"subHeaderWrapper\\" style=\\"display:block;margin-left:auto;margin-right:auto;margin-top:20px;margin-bottom:20px;max-width:500px;font-family:Arial, sans-serif;border-radius:5px;background-color:white\\"><table class=\\"subHeader\\" align=\\"center\\" width=\\"100%\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td><table id=\\"subHeaderTable\\" style=\\"text-align:center;margin-left:auto;margin-right:auto\\"><tr><td><p style=\\"color:gray;font-style:italic;margin-bottom:5px\\">New team</p></td></tr><tr><td><h2 style=\\"font-size:35px;margin-top:0px;margin-bottom:5px\\">Leader</h2></td></tr><tr><td><p>You have been added to Test CPA Team as team leader.</p><div><div></div></div></td></tr></table></td></tr></tbody></table></div><div style=\\"display:block;margin-left:auto;margin-right:auto;text-align:center;width:500px;padding:20px;font-family:Arial, sans-serif;margin-bottom:20px;background-color:white\\"><table class=\\"teamInfo\\" align=\\"center\\" width=\\"500px\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\"><h3 style=\\"margin-top:0px\\">Team Information:</h3></td></tr><tr><td align=\\"left\\"><h4 style=\\"margin-top:0px\\">Name: <span style=\\"font-weight:normal\\">Test CPA Team</span></h4><h4 style=\\"margin-bottom:0px\\">Description:</h4><p style=\\"margin-top:10px\\">This team of CPAs handles all of the tax preparation in the firm.</p></td></tr><tr><td align=\\"left\\"><p>More information about your team can be found by navigating to the Prolaera website.</p><div class=\\"customButton\\" align=\\"center\\" style=\\"max-width:100%;text-align:center;padding:20px 0px 20px 0px\\"><a href=\\"https://app.prolaera.com/#/admin/adminProfileId/dashboard\\" style=\\"width:200px;padding:14px 14px 14px 14px;color:white;text-decoration:none;border-radius:3px\\">View Team</a></div></td></tr></tbody></table></div><table class=\\"emailFooter\\" align=\\"center\\" width=\\"540px\\" cellSpacing=\\"20\\" style=\\"background-color:#333333\\" cellPadding=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td><td style=\\"float:left\\"><span style=\\"font-family:source-sans-pro, sans-serif;font-size:10pt;font-weight:100;line-height:14px;color:#F7F7F7;letter-spacing:1pt;float:left;padding:10px 5px 5px 0px\\">Powered By</span><img alt=\\"react\\" src=\\"https://gallery.mailchimp.com/eb930e30a15249a18d658c183/images/196fa069-3113-464c-a304-a1aa9f99fe22.png\\" width=\\"100\\" height=\\"23\\" style=\\"display:block;outline:none;border:none;text-decoration:none;padding-top:5px\\"/></td><td style=\\"font-family:source-sans-pro, sans-serif;color:#F7F7F7;font-size:10pt;font-weight:100;letter-spacing:1pt;text-align:right;padding-top:10px;padding-right:10px;padding-bottom:10px\\">Â© Prolaera 2018</td></td></tr></tbody></table></tbody></table></td></tr></tbody></table></body></html>"
`;
